cdap-api/src/main/java/co/cask/cdap/api/dataset/DatasetProperties.java

```diff
@@ -69,6 +74,17 @@ protected Builder() {
     }
 
     /**
+     * Sets description of the dataset
+     *
+     * @param description dataset description
+     * @return this builder object to allow chaining
+     */
+    public Builder setDescription(String description) {
+      this.properties.put(DESCRIPTION, description);
```

Comment:
oh, you are adding it as a property. I thought you would add it as an extra field to DatasetProperties. So that you can 
call DatasetProperties.properties() to retrieve all properties, and DatasetProperties.getDescription() to retrieve the description.